### PR TITLE
[ Refactor ] Split TcpServerConnectionManager into itself and TcpServerConnectionManagerBase

### PR DESCRIPTION
**Pull request by lprimak**
Currently class has checkstyle suppression and almost at it's limit of fan-out complexity.
This PR splits this class into a base which handles private computation and it's former self which handles
the public API.

This is needed to add functionality into this class by #17428

### CODE REVIEW

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2444/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2444/Hazelcast-pr-builder-2444.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:checkstyle (default) on project hazelcast: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_2/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_2/hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManager.java:41:8: Unused import - java.util.Iterator. [UnusedImports]
--------------------------
</details>
</pre>

**Comment by nfrankel** [comment]
run-lab-run

**Comment by kwart** [comment]
run-lab-run

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by lprimak** [comment]
@mmedenjak @vbekiaris Have you had a chance to make a decision regarding this?
Which way to go, with this refactor or with `@SuppressWarnings` for fan-out complexity?
Thanks!

**Comment by mmedenjak** [comment]
Hi, I haven't had time to take a look at the PR and I don't know when I'll be able to. @pveentjer can you take a look?

**Comment by lprimak** [comment]
@mmedenjak @pveentjer Did you have any chance to take a look at this?
Pretty simple, all that needs to be done is to decide whether to refactor or suppress the warning.
thank you!

**Comment by Holmistr** [comment]
Hey @lprimak or should I say Hazelcast Hero? :) 

We're currently working hard on getting 4.2 out and that's our top priority. We have a lot of things to be resolved by that time, so we can't make you any promises of merging this PR before 4.2. However, given that it should be rather simple change, I will keep it in the 4.2 scope for now and if we have time, we'll do our best to merge it. We'll keep you posted.

Thanks for another PR, keep it coming! :heart: :beers:

File: hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManager.java [hunk 1]
```diff
@@ -76,87 +58,40 @@
...
 import static java.util.Collections.unmodifiableSet;
 
-@SuppressWarnings("checkstyle:methodcount")
-public class TcpServerConnectionManager
+/**
+ * This is the public APIs for connection manager
+ * Private APIs and supporting stuff is in the Base class
+ */
+public class TcpServerConnectionManager extends TcpServerConnectionManagerBase
         implements ServerConnectionManager, Consumer<Packet>, DynamicMetricsProvider {
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManager.java:71]: `implements ServerConnectionManager` is redundant here (the base class also implements this interface)

**Review by vbekiaris — APPROVED** [review]
looks good, just a minor comment

**Comment by vbekiaris** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManager.java (hunk 1 above)
Comment (lprimak) [reply hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManager.java]: fixed

**Comment by lprimak** [comment]
run-lab-run

**Comment by vbekiaris** [comment]
run-lab-run

**Review by alex-dukhno — APPROVED** [review]

**Comment by mmedenjak** [comment]
Thank you for the refactor @lprimak !

**Pull request merged**